scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/context/TaskContextVariableExtractor.java

```diff
@@ -36,147 +38,342 @@
 import org.ow2.proactive.scheduler.common.task.TaskResult;
 import org.ow2.proactive.scheduler.common.task.TaskVariable;
 import org.ow2.proactive.scheduler.common.task.util.SerializationUtil;
-import org.ow2.proactive.scheduler.common.util.VariableSubstitutor;
 import org.ow2.proactive.scheduler.task.SchedulerVars;
 import org.ow2.proactive.scheduler.task.TaskLauncherInitializer;
 import org.ow2.proactive.scheduler.task.executors.forked.env.ForkedTaskVariablesManager;
 
 
+/**
+ * This class will help you to create a Map containing all variables you need. I will
+ * also manage to resolve values following some constraints of VariableSubstitutor class.
+ */
 public class TaskContextVariableExtractor implements Serializable {
 
     private static final Logger logger = Logger.getLogger(TaskContextVariableExtractor.class);
 
     private final ForkedTaskVariablesManager forkedTaskVariablesManager = new ForkedTaskVariablesManager();
 
-    public Map<String, String> extractVariablesThirdPartyCredentialsAndSystemEnvironmentVariables(
-            TaskContext taskContext) throws Exception {
+    /**
+     * Retrieve all third party credential variables in a map.
+     *
+     * @param taskContext all information to extract third party credentials is here.
+     *
+     * @return map containing thirdPartyCredentials
+     */
+    public Map<String, String>
+            extractVariablesThirdPartyCredentialsAndSystemEnvironmentVariables(TaskContext taskContext) {
         ForkEnvironment forkEnvironment = taskContext.getInitializer().getForkEnvironment();
-        Map<String, Serializable> variables = this.extractVariables(taskContext, true);
-        Map<String, String> thirdPartyCredentials = forkedTaskVariablesManager.extractThirdPartyCredentials(taskContext);
+        Map<String, Serializable> variables = new HashMap<>();
+
+        try {
+            variables = extractAllVariables(taskContext, null, "");
+        } catch (IOException | ClassNotFoundException e) {
+            logger.error("Error reading variables from task context!");
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "Error reading variables from task context!" 6 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)